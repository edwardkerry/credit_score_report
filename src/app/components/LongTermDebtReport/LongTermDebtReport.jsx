import React, { Component } from 'react';
import { number, string } from 'prop-types';

import bemHelper from '../../utils/bem';

const cn = bemHelper({ block: 'report' });

const monthlyChange = (debtChange) => {
  const direction = debtChange >= 0 ? 'Up' : 'Down'
  return `${direction} £${Math.abs(debtChange)} from last month`
}

const LongTermDebtReport = ({ currentCreditLimit, currentDebt, debtChange }) => {
  return (
    <div className={cn('longTermDebt', null)}>
      Your long term debt total 
        <p className={cn('currentDebt', null)}>£{currentDebt}</p>

        <p className={cn('creditLimit', null)}>Total credit limit £{currentCreditLimit} </p>
      <p className={cn('debtChange', null)}>{monthlyChange(debtChange)} </p>
    </div>
  )
}

LongTermDebtReport.propTypes = {
  currentCreditLimit: number.isRequired,
  currentDebt: number.isRequired,
  debtChange: number.isRequired 
}

export default LongTermDebtReport;