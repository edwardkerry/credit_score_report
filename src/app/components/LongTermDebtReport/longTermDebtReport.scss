.report__longTermDebt {
  font-family: Arial, Helvetica, sans-serif;

  .report__currentDebt {
     line-height: 0;
     font-size: 50px;
     color: #dfb7f6;
     @media only screen and (min-device-width: 320px) and (max-device-width: 480px) {
       font-size: 30px;
     }
  }

  .report__debtChangeDown {
    color: #C9DE8B
  }

  .report__debtChangeUp {
    color: rgb(214, 87, 165)
  }
}