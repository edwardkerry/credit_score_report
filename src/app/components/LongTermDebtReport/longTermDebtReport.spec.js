import React from 'react';
import renderer from 'react-test-renderer';
import LongTermDebtReport from './LongTermDebtReport.jsx';

describe('LongTermDebtReport', () => {
  it('renders', () => {

    const currentCreditLimit = 0;
    const currentDebt = 800;
    const debtChange = -100

    const component = renderer.create(
    <LongTermDebtReport 
      currentCreditLimit={currentCreditLimit} 
      currentDebt={currentDebt} 
      debtChange={debtChange}/>
    );
    
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('displays when long term debt has increased', () => {
    const currentCreditLimit = 0;
    const currentDebt = 800;
    const debtChange = 100

    const component = renderer.create(
      <LongTermDebtReport
        currentCreditLimit={currentCreditLimit}
        currentDebt={currentDebt}
        debtChange={debtChange} />
    );

    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('displays when long term debt has decreased', () => {
    const currentCreditLimit = 0;
    const currentDebt = 800;
    const debtChange = -100

    const component = renderer.create(
      <LongTermDebtReport
        currentCreditLimit={currentCreditLimit}
        currentDebt={currentDebt}
        debtChange={debtChange} />
    );

    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
