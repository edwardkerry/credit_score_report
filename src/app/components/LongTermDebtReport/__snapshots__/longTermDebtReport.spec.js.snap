// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LongTermDebtReport displays when long term debt has decreased 1`] = `
<div
  className="longTermDebt__report"
>
  Your long term debt total
  <p
    className="longTermDebt__currentDebt"
  >
    £
    800
  </p>
  <p
    className="longTermDebt__creditLimit"
  >
    Total credit limit £
    0
     
  </p>
  <p
    className="longTermDebt__debtChange"
  >
    Down £100 from last month
     
  </p>
</div>
`;

exports[`LongTermDebtReport displays when long term debt has increased 1`] = `
<div
  className="longTermDebt__report"
>
  Your long term debt total
  <p
    className="longTermDebt__currentDebt"
  >
    £
    800
  </p>
  <p
    className="longTermDebt__creditLimit"
  >
    Total credit limit £
    0
     
  </p>
  <p
    className="longTermDebt__debtChange"
  >
    Up £100 from last month
     
  </p>
</div>
`;

exports[`LongTermDebtReport renders 1`] = `
<div
  className="longTermDebt__report"
>
  Your long term debt total
  <p
    className="longTermDebt__currentDebt"
  >
    £
    800
  </p>
  <p
    className="longTermDebt__creditLimit"
  >
    Total credit limit £
    0
     
  </p>
  <p
    className="longTermDebt__debtChange"
  >
    Down £100 from last month
     
  </p>
</div>
`;
